src/services/inlayHints.ts

```diff
@@ -421,6 +444,258 @@ export function provideInlayHints(context: InlayHintsContext): InlayHint[] {
+                case SyntaxKind.UnknownKeyword:
+                case SyntaxKind.VoidKeyword:
+                case SyntaxKind.ThisType:
+                    parts.push({ text: tokenToString(node.kind)! });
+                    break;
+                case SyntaxKind.Identifier:
+                    const identifier = node as Identifier;
+                    parts.push(getNodeDisplayPart(idText(identifier), identifier));
+                    break;
+                case SyntaxKind.QualifiedName:
+                    const qualifiedName = node as QualifiedName;
+                    visitDisplayPart(qualifiedName.left);
+                    parts.push({ text: "." });
+                    visitDisplayPart(qualifiedName.right);
+                    break;
+                case SyntaxKind.TypePredicate:
+                    const predicate = node as TypePredicateNode;
+                    if (predicate.assertsModifier) {
+                        parts.push({ text: "asserts " });
+                    }
+                    visitDisplayPart(predicate.parameterName);
+                    if (predicate.type) {
+                        parts.push({ text: " is " });
+                        visitDisplayPart(predicate.type);
+                    }
+                    break;
+                case SyntaxKind.TypeReference:
+                    const typeReference = node as TypeReferenceNode;
+                    visitDisplayPart(typeReference.typeName);
+                    if (typeReference.typeArguments) {
+                        parts.push({ text: "<" });
+                        visitDisplayPartList(typeReference.typeArguments, ",");
+                        parts.push({ text: ">" });
+                    }
+                    break;
+                case SyntaxKind.TypeParameter:
+                    // TODO: Modifier list.
+                    const typeParameter = node as TypeParameterDeclaration;
+                    visitDisplayPart(typeParameter.name);
+                    if (typeParameter.constraint) {
+                        parts.push({ text: " extends " });
+                        visitDisplayPart(typeParameter.constraint);
+                    }
+                    if (typeParameter.default) {
+                        parts.push({ text: " = " });
+                        visitDisplayPart(typeParameter.default);
+                    }
+                    break;
+                case SyntaxKind.ConstructorType:
+                    const constructorType = node as ConstructorTypeNode;
+                    parts.push({ text: "new " });
+                    if (constructorType.typeParameters) {
+                        parts.push({ text: "<" });
+                        visitDisplayPartList(constructorType.typeParameters, ",");
+                        parts.push({ text: ">" });
+                    }
+                    // TODO: Parameters.
+                    parts.push({ text: " => " });

```

Comment:
Instead of creating objects on the fly, I'm pretty sure we have display part functions that construct these. They have the benefit that they can include a display part kind as well if I recall correctly.

On that note, I think spaces are typically their own display part.

Reply:
Yeah but those are for `SymbolDisplayPart`s, and I have `InlayHintDisplayPart`s here.